src/adapter/graphql/src/queries/flows/flow_config_snapshot.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use kamu_flow_system::{ConfigSnapshot, Schedule};
+
+use crate::prelude::*;
+
+#[derive(SimpleObject)]
+pub struct FlowConfigurationSnapshot {
+    batching_rule: Option<FlowConfigurationBatching>,
+    schedule: Option<FlowConfigurationSchedule>,
+    compacting_rule: Option<FlowConfigurationCompacting>,
+}
```

Comment:
Can it ever have more than 1 of these fields set to Some(_)? Looks more like an enum to me.
I think a case when all 3 are None is also impossible.

Reply:
Basically for us important to know only one of these configs first my realization was with enum(similar to input for mutation) but I thought that it is possible that in the future we may have some combination of different rules per flow. But I'm totally fine to put here enum for now